### PR TITLE
8323715: Serial: Move genMemoryPools to serial folder

### PR DESCRIPTION
**Pull request by albertnetymk**
Simple moving files from shared to serial folder.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8323715](https://bugs.openjdk.org/browse/JDK-8323715): Serial: Move genMemoryPools to serial folder (**Enhancement** - P4)

### Reviewers
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to [532b5927](https://git.openjdk.org/jdk/pull/17418/files/532b592780b6afd4c7d5d395f1a1f3cb688587cc)
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to [61953838](https://git.openjdk.org/jdk/pull/17418/files/6195383840ba30eca31e17a916faa074315fb930)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17418/head:pull/17418` \
`$ git checkout pull/17418`

Update a local copy of the PR: \
`$ git checkout pull/17418` \
`$ git pull https://git.openjdk.org/jdk.git pull/17418/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17418`

View PR using the GUI difftool: \
`$ git pr show -t 17418`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17418.diff">https://git.openjdk.org/jdk/pull/17418.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17418#issuecomment-1891719168)

### CODE REVIEW

File: src/hotspot/share/gc/serial/genMemoryPools.cpp [hunk 1]
```diff
@@ -24,11 +24,9 @@
 
 #include "precompiled.hpp"
 #include "gc/serial/generation.hpp"
-#include "gc/shared/genMemoryPools.hpp"
+#include "gc/serial/genMemoryPools.hpp"
 #include "gc/shared/space.hpp"
-#if INCLUDE_SERIALGC
 #include "gc/serial/defNewGeneration.hpp"
```
Comment (stefank) [inline src/hotspot/share/gc/serial/genMemoryPools.cpp:29]: The defNewGeneration.hpp line needs to move up.

**Review by stefank — CHANGES_REQUESTED** [review]
Could the files be renamed from genMemoryPools.* to serialMemoryPools.*?

File: src/hotspot/share/gc/serial/serialMemoryPools.hpp [hunk 2]
```diff
@@ -22,8 +22,8 @@
  *
  */
 
-#ifndef SHARE_GC_SHARED_GENMEMORYPOOLS_HPP
-#define SHARE_GC_SHARED_GENMEMORYPOOLS_HPP
+#ifndef SHARE_GC_SERIAL_GENMEMORYPOOLS_HPP
+#define SHARE_GC_SERIAL_GENMEMORYPOOLS_HPP
```
Comment (stefank) [inline src/hotspot/share/gc/serial/serialMemoryPools.hpp:26]: The include guards need to be updated as well.

**Review by stefank — CHANGES_REQUESTED** [review]

**Review by tschatzl — APPROVED** [review]
Lgtm.

Please also update copyright year at your discretion before pushing.

**Review by stefank — APPROVED** [review]

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**